### ISSUE TITLE
Deep Learning and Anomaly Detection

### ISSUE THREAD

**Issue by StankoKuveljic** [ftn-ai-lab/nm-2016#2]
# Smart Alerting

Goal of this project is to make neural network model that will detect type of anomalies on Apache Cassandra nodes. Slow queries on *Apache Cassandra* nodes are caused by some anomalies in CPU, Memory, Disk, Network etc. This project will compare results of *convolutional 1D* and *recurrent neural network*.

## Dataset
I will use labeled data. Labels represent type of anomaly. Monitoring machine will collect time series metrics on Cassandra nodes and store data in database. This data will represent normal type of data. Then, I will use tools to cause anomalies on cluster (cause high cpu usage on nodes and that will get metric with label cpu anomaly. Same is for memory, disk etc).
After collecting dataset, next task is to prepare data for training. Some preprocessing steps will be feature extraction, mean normalization etc. 

## Optional Task
* If I have time untill 16.01.2017
Train model with unlabeled data that will detect just if anomaly happened or not. Value of this task will be constant learning on streaming data, so model will always be „fresh“ and will follow behavior of cluster. And if anomaly is happen, model will send email alert with timestamp and current values of metrics and cause of anomaly (**original task**). On this task I plan to use *autoencoders* and/or *GAN*.

## Tech
Project will use: *[Scala](https://www.scala-lang.org/) (optional [Python](https://www.python.org/)), [TensorFlow](https://www.tensorflow.org/), [Apache Spark](http://spark.apache.org/), [Apache Cassandra](http://cassandra.apache.org/)*

Working **alone** on this project: *Stanko Kuveljić E2 20/2016*

**Comment by jocicmarko** [comment]
Odobrena tema.

Ostavi github repo link u komentaru.
